### ISSUE TITLE
Multiobjective in OptunaSearchCV

### ISSUE THREAD

**Issue by vitorpbarbosa7** [optuna/optuna#3165]
## Motivation

Reduce overfitting in optunaSearchCV

## Description

Allow optimization with directions "maximize" and "minimize" in multiobjective metrics in optunaSearchCV.

Also introduce a feature to optimize the overfitting by reducing the difference between user_attrs_mean_train_score and user_attrs_mean_test_score

**Comment by nzw0301** [comment]
> Allow optimization with directions "maximize" and "minimize" in multiobjective metrics in optunaSearchCV.

Since 1 ) `sklearn.model_selection.RandomizedSearchCV` whose API is imitated by `OptunaSearchCV` supports only a single metric and multi-objective optimisation has multiple solutions, ​so we cannot implement such a feature now, unfortunately.

> Also introduce a feature to optimize the overfitting by reducing the difference between user_attrs_mean_train_score and user_attrs_mean_test_score

To avoid overfitting is a really important thing. However, as a machine learning researcher, I'd like to say that such optimisation might be meaningless because the trivial solution is `user_attrs_mean_train_score = user_attrs_mean_train_score` for any `user_attrs_mean_train_score`. I suggest using train/val splits to avoid overfitting.

**Comment by vitorpbarbosa7** [comment]
Thank you for the reply and suggestions, Kento.
